### PR TITLE
[MAINTENANCE] Reformat core expectation docstrings

### PR DESCRIPTION
**Pull request by kenwade4**
Greatly improve the consistency, format, and usefulness of the individual Expectation detail pages for Expectations on https://greatexpectations.io/expectations

Changes proposed in this pull request:
- Update build_gallery.py to also dump a docstrings.txt file
- Make sure that items under 'Args:' section of docstrings end up as single lines
- Make sure that 'For example' sections of docstrings have the :: in the correct place
- For most expectation docstrings that have a 'Notes:' section, remove the code block and add bullet point
- For kl_divergence and quantile expectations docstring, make sure the complicated 'Notes:' sections are ok
- Update docstrings to use markdown links to current docs
- Fix indent levels and some other minor things

### CODE REVIEW

File: assets/scripts/build_gallery.py [hunk 1]
```diff
@@ -622,12 +633,16 @@ def main(**kwargs):
...
+    docstrings = expectation_docstrings.getvalue()
     if tracebacks != "":
         with open("./gallery-tracebacks.txt", "w") as outfile:
             outfile.write(tracebacks)
     if checklists != "":
         with open("./checklists.txt", "w") as outfile:
             outfile.write(checklists)
+    if docstrings != "":
+        with open("./docstrings.txt", "w") as outfile:
+            outfile.write(docstrings)
```
Comment (Shinnnyshinshin) [inline assets/scripts/build_gallery.py:645]: Thanks for the description in the PR for this, and for the change itself :)

File: great_expectations/expectations/core/expect_column_distinct_values_to_contain_set.py [hunk 2]
```diff
@@ -24,7 +24,43 @@
 
 
 class ExpectColumnDistinctValuesToContainSet(ColumnExpectation):
-    """Expect the set of distinct column values to contain a given set."""
+    """Expect the set of distinct column values to contain a given set.
+
+    expect_column_distinct_values_to_contain_set is a \
+    [Column Aggregate Expectation](https://docs.greatexpectations.io/docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_aggregate_expectations).
+
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/core/expect_column_distinct_values_to_contain_set.py:31]: ❤️

File: great_expectations/expectations/core/expect_column_kl_divergence_to_be_less_than.py [hunk 3]
```diff
@@ -46,135 +46,108 @@
...
+    the difference between the two distributions. A relative entropy of zero indicates that the data are \
+    distributed identically, when binned according to the provided partition.
+
+    In many practical contexts, choosing a value between 0.5 and 1 will provide a useful test.
+
+    This expectation works on both categorical and continuous partitions. See notes below for details.
+
+    expect_column_kl_divergence_to_be_less_than is a \
+    [Column Aggregate Expectation](https://docs.greatexpectations.io/docs/guides/expectations/creating_custom_expectations/how_to_create_custom_column_aggregate_expectations).
+
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/core/expect_column_kl_divergence_to_be_less_than.py:63]: This one looks great, but doesn't seem to be appearing the search results of `Explore Expectations` in our `https://greatexpectations.io/expectations/` site. Could you confirm?

File: great_expectations/expectations/core/expect_column_pair_values_a_to_be_greater_than_b.py [hunk 4]
```diff
@@ -30,30 +32,33 @@ class ExpectColumnPairValuesAToBeGreaterThanB(ColumnPairMapExpectation):
...
+        Exact fields vary depending on the values passed to result_format, include_config, catch_exceptions, and meta.
     """
 
     # This dictionary contains metadata for display in the public gallery
     library_metadata = {
         "maturity": "production",
         "tags": [
             "core expectation",
-            "multi-column expectation",
+            "column pair map expectation",
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/core/expect_column_pair_values_a_to_be_greater_than_b.py:61]: 👍🏼

File: great_expectations/expectations/core/expect_column_values_to_be_dateutil_parseable.py [hunk 5]
```diff
@@ -21,38 +21,34 @@ class ExpectColumnValuesToBeDateutilParseable(ColumnMapExpectation):
...
 
     Args:
         column (str): \
             The column name.
 
     Keyword Args:
         mostly (None or a float between 0 and 1): \
-            Return `"success": True` if at least mostly fraction of values match the expectation. \
-            For more detail, see :ref:`mostly`.
+            Successful if at least mostly fraction of values match the expectation. \
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/core/expect_column_values_to_be_dateutil_parseable.py:32]: I know it's a small wording change, but I do appreciate it. Thank you

**Review by Shinnnyshinshin — APPROVED** [review]
@kenwade4 Thank you so much for tackling this beast. LGTM to me, and thank you for getting me acquainted with [the grip tool](https://github.com/joeyespo/grip) in the process as well :) ✅

File: great_expectations/expectations/core/expect_column_kl_divergence_to_be_less_than.py (hunk 3 above)
Comment (kenwade4) [reply great_expectations/expectations/core/expect_column_kl_divergence_to_be_less_than.py:63]: That's correct... it's hardcoded to skip for now https://github.com/great-expectations/great_expectations/blob/develop/assets/scripts/build_gallery.py#L298-L303

**Pull request merged**